code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -288,14 +288,17 @@
 
 /mob/living/silicon/ai/can_interact_with(atom/A)
 	. = ..()
+	var/turf/ai = get_turf(src)
+	var/turf/target = get_turf(A)	
 	if (.)
 		return
+	if ((ai.z != target.z) && !is_station_level(ai))

```

Comment:
this locks it to the station z level, I thought you were just restricting it only to it's own z.

Reply:
Did I write it incorrectly? It should play out as AI being able to control gulag and such from the station, but not allow an AI on the gulag to control the stations z level, while still allowing the gulag AI to control the gulag.

Reply:
you wrote this in line with how you want it to function.

If the AI is offstation AND it tries to click on something off it's current Z it will not work.